ec2/tests/test_ec2.py

```diff
@@ -24,11 +24,14 @@
 from ec2 import connection
 from ec2 import instance
 from ec2 import utils
+from ec2 import floatingip
 from cloudify.mocks import MockCloudifyContext
+from cloudify.mocks import MockContext
 from cloudify.exceptions import NonRecoverableError
 
 TEST_AMI_IMAGE_ID = 'ami-e214778a'
 TEST_INSTANCE_TYPE = 't1.micro'
+FQDN = '((?:[a-z][a-z\\.\\d\\-]+)\\.(?:[a-z][a-z\\-]+))(?![\\w\\.])'
 
 
 class TestPlugin(testtools.TestCase):
```

Comment:
You should have different classes for testing different resources.

Maybe something like:

```python
class TestInstance(...)
   ...

class TestElasticip(...)
   ...